### DISCUSSION TITLE
data importer setup with token prompts ID and URL

### DISCUSSION THREAD

**Discussion by anoderay** [firefly-iii/firefly-iii#7059] [Firefly III]
I struggle to get FIDI authentication to work as I at least understand it _should_ work. Importing_ csv files works fine but I seem to always have to re-authenticate using a client ID and URL..which from what I understand shouldn't be the case? 

I supply the following via the env-file:
FIREFLY_III_URL=http://172.22.0.5:8080 (the internal docker ID portainer shows me for the Firefly-Container)
VANITY_URL=http://192.168.1.150:8080 (which I can use to access Firefly in my local network)
FIREFLY_III_ACCESS_TOKEN=[gibberish]

But when I call 192.168.1.150:8081 to access FIDI I am prompted to provide User ID and URL.

Also I think in the [documentation](https://docs.firefly-iii.org/data-importer/install/configure/) _"To authenticate with Firefly III you must set ONE the following variables:"_ could be rephrased as "To authenticate with Firefly III you must set EITHER ONE of the following variables". (at least an "of" is missing in that sentence :-)).

**Comment by JC5** [comment]
What happens when you press [Reauthenticate]?

**Reply by anoderay** [reply]
I get prompted for User ID and URL.

**Reply by JC5** [reply]
Alright, can you share the logs for me, from the moment you press [Reauthenticate]?

**Reply by anoderay** [reply]
Looking at the logs I noticed that the error came from my docker configuration. FIDI simply wasnt pulling the correct credentials from the env-file because I didn't tell it to do so in the compose file (...). Curiously though: if I use a combined env-file for firefly and FIDI I simply get a "500 Server Error". Don't know whether this is expected?

**Reply by JC5** [reply]
That depends on the 500 error, do you know what's in the logs about it?
